src/components/header/Header.tsx

```diff
@@ -1,9 +1,41 @@
+import { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
 import './Header.css';
 
 export const Header = () => {
+  const [isVisible, setIsVisible] = useState(true);
+  const [pastThreshold, setPastThreshold] = useState(false);
+
+  useEffect(() => {
```

Comment:
Great idéa with the menu! Makes the site much better when you have the menu available like that